### PR TITLE
icdc_2381 - Display variant tracks

### PR DESCRIPTION
**Pull request by rana22**
1. Jbrowse version update
2. dispaly varient track

### New Features
1. Jbrowse version update
2. display variant track

### CODE REVIEW

**Review by iksheth — APPROVED** [review]

File: src/bento/JBrowseData.js [hunk 1]
```diff
@@ -22,12 +26,118 @@ export const varient = {
   trackId: 'my_varient_track',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:26]: varient should be variant

File: src/bento/JBrowseData.js [hunk 2]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
   trackName: 'My Varient',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:27]: Varient should be Variant

File: src/bento/JBrowseData.js [hunk 3]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
+    {
+      display: false,
+      trackId: 'ReferenceSequenceTrack',
+      name: 'Reference Sequence',
+      assemblyNames: [...assemblyNames],
+      type: 'FeatureTrack',
+      adapter: {
+        type: 'BgzipFastaAdapter',
+        fastaLocation: {
+          uri: 'https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:48]: This is not correct. This is a uri for the human reference genome.

File: src/bento/JBrowseData.js [hunk 4]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
+          locationType: 'UriLocation',
+        },
+        faiLocation: {
+          uri: 'https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz.fai',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:52]: Same. This is the human index.

File: src/bento/JBrowseData.js [hunk 5]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
+        },
+        gziLocation: {
+          uri: 'https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz.gzi',
+          locationType: 'UriLocation',
+        },
+      },
+    },
+    {
+      display: false,
+      trackId: 'repeats_hg19',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:63]: We should change this to 'repeats_canFam'

File: src/bento/JBrowseData.js [hunk 6]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
+      type: 'FeatureTrack',
+      display: false,
+      trackId: 'gff3tabix_genes',
+      assemblyNames: [...assemblyNames],
+      name: 'GFF3Tabix genes',
+      category: ['Miscellaneous'],
+      adapter: {
+        type: 'Gff3TabixAdapter',
+        gffGzLocation: {
+          uri: 'volvox.sort.gff3.gz',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:86]: This is a gff3 file for volvox, we need the gff3 for CanFam

File: src/bento/JBrowseData.js [hunk 7]
```diff
@@ -22,12 +26,118 @@ export const varient = {
...
+          locationType: 'UriLocation',
+        },
+        index: {
+          location: {
+            uri: 'volvox.sort.gff3.gz.tbi',
```
Comment (kuffelgr) [inline src/bento/JBrowseData.js:91]: Same, this is for volvox, not CanFam

File: src/pages/JbrowseDetail/JbrowseDetailView.js [hunk 8]
```diff
@@ -39,31 +44,50 @@ const getAdapter = ({ bamLocationUri, indexUri }) => {
...
+    UriLocation,
+  );
+  const varientIndex = new Index(new FileLocation(
+    indexUri,
+    UriLocation,
+  ));
+  const adapter = {
+    type: VarientAdapter,
+    vcfGzLocation: varFileLocation,
+    index: varientIndex,
```
Comment (kuffelgr) [inline src/pages/JbrowseDetail/JbrowseDetailView.js:59]: varient should be variant

File: src/pages/JbrowseDetail/JbrowseDetailView.js [hunk 9]
```diff
@@ -39,31 +44,50 @@ const getAdapter = ({ bamLocationUri, indexUri }) => {
...
+        session.view.tracks.push({ ...viewTrack });
+      }
     });
   }
   return session;
 };
 
 const getTracks = ({
-  alignments, alignmentUris,
+  alignmentUris, varientUris, optionalTracks,
```
Comment (kuffelgr) [inline src/pages/JbrowseDetail/JbrowseDetailView.js:87]: varient should be variant

File: src/pages/JbrowseDetail/JbrowseDetailView.js [hunk 10]
```diff
@@ -75,23 +99,40 @@ const getTracks = ({
     );
     allTracks.push(alignmentOpts);
   }
+
+  if (varientUris.file_name) {
+    const varientAdapter = getVarient(varientUris);
+    const variantOpts = new Track(
+      varient.trackId,
```
Comment (kuffelgr) [inline src/pages/JbrowseDetail/JbrowseDetailView.js:106]: varient should be variant

File: src/pages/JbrowseDetail/JbrowseDetailView.js [hunk 11]
```diff
@@ -103,8 +144,21 @@ const JBrowseViewDetail = ({
       });
     }
 
+    const varientUris = {};
```
Comment (kuffelgr) [inline src/pages/JbrowseDetail/JbrowseDetailView.js:147]: varient should be variant

File: src/pages/JbrowseDetail/JbrowseDetailView.js [hunk 12]
```diff
@@ -103,8 +144,21 @@ const JBrowseViewDetail = ({
...
+    if (vcfFiles.length > 0) {
+      vcfFiles.forEach((file) => {
+        varientUris.file_name = file.file_name;
```
Comment (kuffelgr) [inline src/pages/JbrowseDetail/JbrowseDetailView.js:150]: varient should be variant

**Pull request merged**